QEfficient/cloud/infer.py

```diff
@@ -45,102 +44,166 @@ def main(
-            mos=mos,
-            device_group=device_group,
-        )
-        assert (
-            generated_qpc_path == qpc_dir_path
-        ), f"QPC files were generated at an unusual location, expected {qpc_dir_path}; got {generated_qpc_path}"
-        cloud_ai_100_exec_kv(
+        compile_and_execute_onnx(
+            onnx_model_path,
+            tokenizer,
             batch_size,
-            tokenizer=tokenizer,
-            qpc_path=qpc_dir_path,
-            device_id=device_group,
-            prompt=prompt,
+            qpc_dir_path,
+            num_cores,
+            prompt_len,
+            ctx_len,
+            mxfp6,
+            mxint8,
+            aic_enable_depth_first,
+            mos,
+            device_group,
+            prompt,
         )
         return
 
-    #############################################
-    # hf model -> export -> compile -> execute
-    #############################################
+    # If none of the above conditions met, proceed with downloading, transforming, exporting, compiling, and executing
     model_hf = AutoModelForCausalLM.from_pretrained(model_hf_path, use_cache=True)
-    # Easy and minimal api to update the model to QEff.
     model_transformed = QEfficient.transform(model_hf, type="Transformers", form_factor="cloud")
-    logger.info(f"Model after Optimized transformations {model_transformed}")
-
-    # Export to the Onnx
-    logger.info(f"Exporting to Pytorch {model_name} to ONNX...")
-    base_path, generated_onnx_path = qualcomm_efficient_converter(
-        model_kv=model_transformed,
-        onnx_dir_path=onnx_dir_path,
-        model_name=model_name,
-        kv=True,
-        form_factor="cloud",
-        return_path=True,
-        tokenizer=tokenizer,
+    onnx_model_path = export_to_onnx(model_transformed, model_name, tokenizer, onnx_dir_path, onnx_model_path)
+    compile_and_execute_onnx(
+        onnx_model_path,
+        tokenizer,
+        batch_size,
+        qpc_dir_path,
+        num_cores,
+        prompt_len,
+        ctx_len,
+        mxfp6,
+        mxint8,
+        aic_enable_depth_first,
+        mos,
+        device_group,
+        prompt,
     )
-    print(
-        f"Generated Onnx_path {generated_onnx_path} and Onnx_model_path {onnx_model_path} and Onnx_dir_path is {onnx_dir_path}"
+
+
+def create_qpc_base_dir_name(
+    num_cores: int,
+    batch_size: int,
+    prompt_len: int,
+    ctx_len: int,
+    mos: int,
+    device_count: int,
+    mxfp6: bool,
+    mxint8: bool,
+) -> str:
+    """
+    Create the base directory name for the QPC based on the parameters.
+
+    Returns:
+    - A string representing the base directory name.
+    """
+    return f"qpc_{num_cores}cores_{batch_size}BS_{prompt_len}PL_{ctx_len}CL_{mos}MOS_{device_count}devices{'_mxfp6_mxint8' if mxfp6 and mxint8 else '_mxfp6' if mxfp6 else '_fp16_mxint8' if mxint8 else '_fp16'}"
+
+
+def process_prompt(prompt: str, prompts_txt_file_path: str, batch_size: int) -> str:
```

Comment:
just one liner wrapper?

Reply:
Just for neatness, keeping main simple and neat. (better readability)

Reply:
yes, this works too.